src/app/ToDo/Components/to-do.component.ts

```diff
@@ -1,66 +1,92 @@
 import { Component, OnInit } from '@angular/core';
+import {
+  TodosState
+} from '../state'
+import * as TodoActions from '../state/todo.action'
+import { ToDo } from '../state/todo.model';
+import { Store } from '@ngrx/store';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { ToDoFacade } from '../state/todo.facade';
 import { Observable, Subscription } from 'rxjs';
-import { map } from 'rxjs/operators';
-import ToDo from '../state/todo.model';
-import ToDoState from '../state/todo.state';
-import { ToDoFacade } from './../state/todo.facade';
 
 @Component({
   selector: 'app-to-do',
-  templateUrl: './to-do.component.html',
-  styleUrls: ['./to-do.component.css']
+  templateUrl: './to-do.component.html'
 })
 export class ToDoComponent implements OnInit {
-  todo$: Observable<ToDoState>;
-  ToDoSubscription: Subscription;
-  ToDoList: ToDo[] = [];
 
-  Title: string = '';
-  IsCompleted: boolean = false;
-  todoError: Error = null;
+  public todo$: Observable<any>;
+  public todoForm: FormGroup;
+  public toDoSubscription: Subscription;
 
-  constructor(private _todoFacade: ToDoFacade) {
+  public loadTask(task: ToDo) {
+    this.todoForm.patchValue(task);
+  }
+
+  constructor(
+    private _todoFacade: ToDoFacade,
```

Comment:
Please use `readonly` where possible.
Additionally, we do not follow the old convention of naming private variables.
We only use the old convention when we use `set` and `get`.

Reply:
Hmm, cool. In this case, would be "todoFacade" the name?

Reply:
Yup